backend/src/app.ts

```diff
@@ -1,9 +1,24 @@
-import express from "express";
+import express, { Request, Response } from "express";
 import mongoose from "mongoose";
 import { json } from "body-parser";
 import { userRouter } from "./routes/user";
 import { sessionsRouter } from "./routes/sessions";
 import { mongoURI, port } from "./config";
+import { ValidationError, CustomError } from "./errors/index";
+
+/**
+ * Generic Error Handler
+ */
+const errorHandler = (err: CustomError, _req: Request, res: Response) => {
```

Comment:
Added a generic error handler for all our routes. We can keep on updating this going forward so that we don't have to repeat code in each of the routes. 

To understand how the handler is working (for someone with less backend experience as well) - 
1. Understand what express middlewares are - https://www.youtube.com/watch?v=lY6icfhap2o
2. In the sequence of middlewares hit for any request on the backend, the errorHandler middleware is the last one. If there was an error, the previous middleware sends some error to this middleware so err is not null. 
3. If everything went smoothly, this middleware won't generally be triggered because all routes directly send a response if successful. 

please let me know if you have trouble understanding the middleware concept because that is pretty important in express programming.